File: frontend/src/types/prediction.ts
export type PredictionInput = { location: string; carpet_area_sqft: number; floor_num: number; bathroom: number; balcony: number; car_parking: number; furnishing: string; transaction: string; ownership: string; facing: string };
export type PredictionResult = { predicted_price: number; currency: string };

File: backend/app/schemas/prediction.py
from pydantic import BaseModel, Field


class PredictionRequest(BaseModel):
    location: str = Field(min_length=1, max_length=120)
    carpet_area_sqft: float = Field(gt=0, le=100_000)
    floor_num: int = Field(ge=-5, le=200)
    bathroom: int = Field(ge=0, le=30)
    balcony: int = Field(ge=0, le=30)
    car_parking: int = Field(ge=0, le=30)
    furnishing: str = Field(min_length=1)
    transaction: str = Field(min_length=1)
    ownership: str = Field(min_length=1)
    facing: str = Field(min_length=1)


class PredictionResponse(BaseModel):
    predicted_price: float
    currency: str = "INR"
